src/main/java/org/cirdles/topsoil/data/SimpleDataset.java

```diff
@@ -46,4 +57,20 @@ public void setName(String name) {
         this.name = Optional.ofNullable(name);
     }
 
+    @Override
+    public void addObserver(Chart chart, VariableContext variableContext) {
+        observers.put(chart, variableContext);
+        entries.stream().forEach(entry -> {
+           entry.addObserver(this);
+        });
+    }
+    
+    @Override
+    public void updateData() {
+        for (Map.Entry pair : observers.entrySet()) {
```

Comment:
This could be `Map.Entry<Chart, VariableContext>`, if I'm not mistaken. That would eliminate the casts below.